x-pack/plugins/cases/public/components/all_cases/utils/all_cases_url_state_deserializer.test.ts

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CustomFieldTypes } from '../../../../common/types/domain';
+import { DEFAULT_CASES_TABLE_STATE } from '../../../containers/constants';
+
+import { allCasesUrlStateDeserializer } from './all_cases_url_state_deserializer';
+import { parseUrlParams } from './parse_url_params';
+
+describe('allCasesUrlStateDeserializer', () => {
+  const { customFields, ...filterOptionsWithoutCustomFields } =
+    DEFAULT_CASES_TABLE_STATE.filterOptions;
+
+  const defaultMap = {
+    ...filterOptionsWithoutCustomFields,
+    ...DEFAULT_CASES_TABLE_STATE.queryParams,
+  };
+
+  it('parses defaults correctly', () => {
+    expect(allCasesUrlStateDeserializer(defaultMap)).toMatchInlineSnapshot(`
+      Object {
+        "filterOptions": Object {
+          "assignees": Array [],
+          "category": Array [],
+          "owner": Array [],
+          "reporters": Array [],
+          "search": "",
+          "searchFields": Array [
+            "title",
+            "description",
+          ],
+          "severity": Array [],
+          "status": Array [],
+          "tags": Array [],
+        },
+        "queryParams": Object {
+          "page": 1,
+          "perPage": 10,
+          "sortField": "createdAt",
+          "sortOrder": "desc",
+        },
+      }
+    `);
+  });
+
+  it('parses an empty object correctly', () => {
+    expect(allCasesUrlStateDeserializer({})).toMatchInlineSnapshot(`
+      Object {
+        "filterOptions": Object {},
+        "queryParams": Object {},
+      }
+    `);
+  });
+
+  it('does not return unknown values', () => {
+    // @ts-expect-error: testing unknown values
+    expect(allCasesUrlStateDeserializer({ foo: 'bar' })).toMatchInlineSnapshot(`
+      Object {
+        "filterOptions": Object {},
+        "queryParams": Object {},
+      }
+    `);
+  });
+
+  it('converts page to integer correctly', () => {
+    // @ts-expect-error: testing integer conversion
+    expect(allCasesUrlStateDeserializer({ page: '1' }).queryParams.page).toBe(1);
+  });
+
+  it('sets perPage to the maximum allowed value if it is set to over the limit', () => {
+    // @ts-expect-error: testing integer conversion
+    expect(allCasesUrlStateDeserializer({ perPage: '1000' }).queryParams.perPage).toBe(100);
+  });
+
+  it('converts perPage to integer correctly', () => {
+    // @ts-expect-error: testing integer conversion
+    expect(allCasesUrlStateDeserializer({ perPage: '2' }).queryParams.perPage).toBe(2);
+  });
+
+  it('sets the defaults to page and perPage correctly if they are not numbers', () => {
+    const url = 'page=foo&perPage=bar';
```

Comment:
super nit: if we moved to Rison maybe that should become the default in tests too

Reply:
`allCasesUrlStateDeserializer` accepts an object so I used that as an input. The URLs are tested in `parseUrlParams`.